src/snowflake/connector/connection_diagnostic.py

```diff
@@ -0,0 +1,562 @@
+        self.test_results: dict[str, Any] = {'INITIAL': [], 'PROXY': [], 'SNOWFLAKE_URL': [], 'STAGE': [],
+                                             'OCSP_RESPONDER': [], 'OUT_OF_BAND_TELEMETRY': [], 'IGNORE': []}
+        host_type: str = "INITIAL"
+        self.__append_message(host_type, f"Specified snowflake account: {self.account}")
+        self.__append_message(host_type, f"Host based on specified account: {self.host}")
+        if '.com.snowflakecomputing.com' in self.host:
+            self.host = host.split('.com.snow')[0] + ".com"
+            logger.warning(f"Account should not have snowflakecomputing.com in it. You provided {host}.  "
+                           f"Continuing with fixed host.")
+            self.__append_message(host_type,
+                                  f"We removed extra .snowflakecomputing.com and will continue with host: "
+                                  f"{self.host}")
+        else:
+            self.host = host
+
+        self.ocsp_urls: list[str] = []
+        self.crl_urls: list[str] = []
+        self.cert_info: dict[str, dict[str, Any]] = {}
+        self.proxy_type: str = "params"
+        self.proxy_host = proxy_host
+        self.proxy_port = proxy_port
+        self.proxy_user = proxy_user
+        self.proxy_password = proxy_password
+        if self.proxy_host is None:
+            proxy_url = os.getenv("HTTPS_PROXY")
+            self.proxy_type = "environment"
+        else:
+            proxy_url = getproxies()['https']
+            self.proxy_type = "system"
+
+        self.proxy_host, self.proxy_port, self.proxy_user, self.proxy_password = self.__parse_proxy(proxy_url)
+        self.__https_host_report(self.host)
+        self.full_connection_diag_log_path: Path | None = (
+            Path(connection_diag_log_path)
+            if connection_diag_log_path is not None
+            else None
+        )
+        self.full_connection_diag_whitelist_path: Path | None = (
+            Path(connection_diag_whitelist_path)
+            if connection_diag_whitelist_path is not None
+            else None
+        )
+        self.tmpdir: str = (
+            tempfile.gettempdir() if IS_WINDOWS
+            else "/tmp"
+        )
+        if self.full_connection_diag_log_path is None:
+            self.full_connection_diag_log_path = Path(self.tmpdir)
+        else:
+            if not self.full_connection_diag_log_path.is_absolute():
+                logger.warning(f"Path {self.full_connection_diag_log_path} for connection test is not absolute.")
+                self.full_connection_diag_log_path = Path(self.tmpdir)
+            elif not self.full_connection_diag_log_path.exists():
+                logger.warning(f"Path {self.full_connection_diag_log_path} for connection test does not exist.")
+                self.full_connection_diag_log_path = Path(self.tmpdir)
+
+        self.report_file: Path = Path.joinpath(self.full_connection_diag_log_path, "SnowflakeConnectionTestReport.txt")
```

Comment:
This works, I just wanted to point out that with pathlib has some nice syntactic sugar in it
```suggestion
        self.report_file: Path = self.full_connection_diag_log_path / "SnowflakeConnectionTestReport.txt"
```